File: civitasRuby/pkg/civitasRuby-0.0.1/lib/civitas/sorpresa_por_jugador.rb
#encoding:utf-8
=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

module Civitas
  
class SorpresaPorJugador < Sorpresa
  
  def initialize (valor, texto)
    super("*Por jugador*", nil, valor, texto, nil)
  end
  
  def aplicarAJugador (actual, todos)
    if jugadorCorrecto(actual, todos)
      informe(actual, todos)

      i = 0
      while i < todos.length
        if i != actual
          todos[i].paga(@valor)
        end
        i += 1
      end
      todos[actual].recibe(@valor)
    end
  end
  
end

end

File: civitasJava/src/civitas/SorpresaPorJugador.java

package civitas;

import java.util.ArrayList;

public class SorpresaPorJugador extends Sorpresa {
    
    SorpresaPorJugador (int valor, String texto) {
        super ("*Por jugador*");
        this.texto = texto;
        this.valor = valor;
    }
    
    /** Todos los jugadores dan/reciben dinero del jugador actual. */
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe (actual, todos);

            for (int i=0 ; i<todos.size() ; i++)
                if (i != actual)
                    todos.get(i).paga(valor);     
            
            todos.get(actual).recibe (valor);
        }
    }
    
}
